airflow/utils/process_utils.py

```diff
@@ -184,3 +185,26 @@ def kill_child_processes_by_pids(pids_to_kill: List[int], timeout: int = 5) -> N
             log.info("Killing child PID: %s", child.pid)
             child.kill()
             child.wait()
+
+
+@contextmanager
+def patch_environ(new_env_variables: Dict[str, str]):
```

Comment:
We already have `temporary_environment_variable` in `google.utils.credentials_provider`. Let's decide to use only one function.

Reply:
I removed  google.utils.credentials_provider.temporary_environment_variable method.